### PR TITLE
emit event_type_published and event_type_subscribed metrics

### PR DESCRIPTION
**Pull request by raypinto**
**Description**

Changes proposed in this pull request:

- Emit metric `event_type_published` when an event is published
- Emit metric `event_type_subscribed` when a new subscription is created
- Create a dashboard mapping the two metrics
- Updated the `delivery per subscription` dashboard - added filters based on eventTypes
- Update troubleshooting documentation 

**Related issue(s)**

### CODE REVIEW

File: components/event-publisher-proxy/pkg/metrics/collector.go [hunk 1]
```diff
@@ -64,3 +78,8 @@ func (c *Collector) RecordError() {
 func (c *Collector) RecordLatency(duration time.Duration) {
 	c.latency.WithLabelValues().Observe(duration.Seconds())
 }
+
+// RecordEventType records a eventType metric
+func (c *Collector) RecordEventType(eventType, eventSource string) {
+	c.eventType.WithLabelValues(eventType, eventSource).Inc()
```
Comment (mfaizanse) [inline components/event-publisher-proxy/pkg/metrics/collector.go:84]: What happens to the metric EventType when EPP is restarted/crashed.
Does it starts counting from zero again?

File: components/eventing-controller/pkg/handlers/metrics/collector.go [hunk 2]
```diff
@@ -51,4 +67,10 @@ func (c *Collector) RecordDeliveryPerSubscription(subscriptionName, eventType, s
 // RegisterMetrics registers the metrics
 func (c *Collector) RegisterMetrics() {
 	metrics.Registry.MustRegister(c.deliveryPerSubscription)
+	metrics.Registry.MustRegister(c.eventTypes)
+}
+
+// RecordEventTypes records the event_type_subscribed metric
+func (c *Collector) RecordEventTypes(subscriptionName, subscriptionNamespace, eventType, consumer string) {
+	c.eventTypes.WithLabelValues(subscriptionName, subscriptionNamespace, eventType, consumer).Inc()
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/handlers/metrics/collector.go:75]: same, What happens to the metrics when EC is restarted/crashed.
Does it starts counting from zero again?

File: tests/fast-integration/eventing-test/nats-dashboard-queries.json [hunk 3]
```diff
@@ -102,5 +102,9 @@
   {
     "title":"NATS JetStream: Message Acks Pending",
     "query":"sum(nats_consumer_num_ack_pending{pod=~\"$server\",stream_name=~\"$stream\",consumer_name=~\"$consumer\"}) by (consumer_name)"
+  },
+  {
+    "title":"NATS JetStream: Event Types Summary",
+    "query":"event_type_subscribed"
```
Comment (mfaizanse) [inline tests/fast-integration/eventing-test/nats-dashboard-queries.json:108]: Have you also added `event_type_published` to fast-integration dashboard tests?

**Review by klaudiagrz — DISMISSED** [review]

File: components/event-publisher-proxy/pkg/metrics/collector.go [hunk 4]
```diff
@@ -11,16 +11,21 @@ const (
 	Errors = "event_publish_to_messaging_server_errors_total"
 	// Latency name of the latency metric
 	Latency = "event_publish_to_messaging_server_latency"
+	// EventType name of the eventType metric
+	EventType = "event_type_published"
```
Comment (mfaizanse) [inline components/event-publisher-proxy/pkg/metrics/collector.go:15]: (nit) would it be better to have little more descriptive name `eventTypePubliushedMetricKey` and `eventTypePubliushedMetricHelp` ?

File: components/event-publisher-proxy/pkg/metrics/metricstest/metricstest.go [hunk 5]
```diff
@@ -18,6 +18,11 @@ func EnsureMetricLatency(t *testing.T, collector *metrics.Collector) {
 	ensureMetricCount(t, collector, metrics.Latency, 1)
 }
 
+// EnsureMetricEventType ensures metric eventType exists
+func EnsureMetricEventType(t *testing.T, collector *metrics.Collector) {
```
Comment (mfaizanse) [inline components/event-publisher-proxy/pkg/metrics/metricstest/metricstest.go:22]: (nit) `EnsureMetricEventTypePublished(` ?

**Review by mfaizanse — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**